test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -1446,6 +1446,49 @@ record = corfuRuntime.getTableRegistry().getRegistryTable().get(tableNameKey);
         assertThat(options.getSecondaryKey(0).getIndexPath()).isEqualTo("event_time");
     }
 
+    /**
+     * Ensure that {@link ObjectID}'s type is used during object cache operations.
+     */
+    @Test
+    public void testTableType() throws Exception {
+        // Get a Corfu Runtime instance.
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+
+        // Creating Corfu Store using a connected corfu client.
+        CorfuStoreShim shimStore = new CorfuStoreShim(corfuRuntime);
+
+        // Define a namespace for the table.
+        final String someNamespace = "some-namespace";
+        // Define table name.
+        final String tableName = "DiskTable";
+        final String dataPath = Files.createTempDirectory(tableName).toString();
+
+        PersistenceOptions persistenceOptions = PersistenceOptions.newBuilder()
+                .setDataPath(dataPath)
+                .build();
+
+        final Table<?, ?, ?> inMemoryTable = shimStore.openTable(
+                someNamespace,
+                tableName,
+                SampleSchema.Uuid.class,
+                SampleSchema.SampleTableAMsg.class,
+                ManagedResources.class,
+                TableOptions.builder().build());
+        assertThat(inMemoryTable.getUnderlyingType()).isEqualTo(PersistentCorfuTable.class);
```

Comment:
PersistentCorfuTable v/s PersistedCorfuTable types. Curious if we have some document to know more about these table types.